DopplerCurrencyJob/DopplerCurrencyService/DopplerCurrencyService.cs

```diff
@@ -173,5 +171,35 @@ private string CreateJwtToken()
 
             return _tokenHandler.WriteToken(jwtToken);
         }
+
+        private async Task<CurrencyResponse> GetPreviousPrices(DateTime cstTime, string currencyCode)
+        {
+            var currentDate = cstTime;
+            CurrencyResponse result = null;
+            var count = 0;
+
+            while (count < _dopplerCurrencySettings.HolidayRetryCountLimit)
```

Comment:
Why don't use `for` sentence in place the `while`?

Reply:
Because I reused the existing code but is ok is you want that I use  `for`